moped-database/views/component_arcgis_online_view.sql

```diff
@@ -202,14 +203,25 @@ SELECT
     plv.knack_project_id AS knack_data_tracker_project_record_id,
     plv.project_url,
     (plv.project_url || '?tab=map&project_component_id='::text) || mpc.project_component_id::text AS component_url,
-    plv.project_development_status,
-    plv.project_development_status_date,
-    plv.project_development_status_date_calendar_year,
-    plv.project_development_status_date_calendar_year_month,
-    plv.project_development_status_date_calendar_year_month_numeric,
-    plv.project_development_status_date_calendar_year_quarter,
-    plv.project_development_status_date_fiscal_year,
-    plv.project_development_status_date_fiscal_year_quarter,
+    CASE
+        WHEN current_phase.phase_name_simple = 'Complete'::text THEN 'Complete'::text
+        WHEN coalesce(mpc.completion_date, plv.substantial_completion_date) IS NOT null
+            THEN
+                CASE
+                    WHEN plv.current_phase_simple = 'Construction'::text THEN 'Estimated End Date (In Construction)'::text
+                    ELSE 'Estimated End Date'::text
+                END
+        WHEN lpmd.latest IS NOT null THEN 'Estimated Public Meeting Date'::text
+        WHEN eaocpd.earliest IS NOT null THEN 'Estimated Start of Project Development'::text
+        ELSE null::text
+    END AS project_development_status,
+    get_project_development_status_date(lpmd.latest::timestamp with time zone, eaocpd.earliest, mpc.project_id, coalesce(mpc.completion_date, plv.substantial_completion_date), plv.substantial_completion_date_estimated, current_phase.phase_name_simple)::text AS project_development_status_date,
```

Comment:
I went with a function here to calculate `project_development_status_date` since the next several fields in this view are different formats of the same value. If anyone can think of a better way to approach this than calling this function for each field, I'd love to hear about it. I wish that we could calculate based on other fields in the same SELECT.

Reply:
thanks for throwing some thoughts about this! very handy since you've been in SQL mode for a while with VZ 🙏 💪  Out of curiosity, I tested out calling the status date function for all the other status date fields and saw about a 100-150ms increase in query time on the full ~12000 rows. I'm going to keep thinking about this for the next issue because I agree that it isn't ideal - especially since there are a bunch of args to pass repeatedly. Just would look kinda yuck.

I'm also taking into account that we will rarely query the full dataset now that we have incremental updates, but I don't know if that is a really good reason to settle and not find a better way. 🤔

Reply:
sorry, i missed this comment from you. yes, 150ms is for sure not a big deal, especially because no human is sitting around waiting for this view to load. maybe a different story if this was powering a UI component.

anyway, i came back to this thread because this exact situation _might_  be a case for a lateral join ([docs](https://www.postgresql.org/docs/current/queries-table-expressions.html#QUERIES-LATERAL)), since the lateral join allows you to "reference columns provided by preceding FROM items."

probably not worth the time to try and tune this any further but i did want to drop that here 👍

Reply:
i was staring at the docs trying to understand that perk of lateral joins the other day when looking at VZ code. this made it click, and i get it now! thanks, John!!!